# HG changeset patch
# User Rik <rik@octave.org>
# Date 1396021675 25200
#      Fri Mar 28 08:47:55 2014 -0700
# Node ID 348f67886dc4d2dedca7efb05826e79db2895308
# Parent  29f00c0d0657a087e094c71bb39c4b27202f7eff
hgload.m: Check for file with .ofig extension before bare file name (bug #41978).

* hgload.m: Check for file with .ofig extension before bare file name.

diff --git a/scripts/plot/util/hgload.m b/scripts/plot/util/hgload.m
--- a/scripts/plot/util/hgload.m
+++ b/scripts/plot/util/hgload.m
@@ -28,28 +28,31 @@
 ## Author: Massimiliano Fasi
 
 function h = hgload (filename)
 
   ## Check input arguments
   if (nargin != 1)
     print_usage ();
   endif
-  
+
   ## Check file existence
-  if (isempty (file_in_loadpath (filename)))
-    [~, ~, ext] = fileparts (filename);
-    if (isempty (ext))
+  [~, ~, ext] = fileparts (filename);
+  if (isempty (ext))
+    if (! isempty (file_in_loadpath ([filename ".ofig"])))
       filename = [filename ".ofig"];
+    elseif (isempty (file_in_loadpath (filename)))
+      error ("hgload: unable to locate file %s", filename);
     endif
+  else
     if (isempty (file_in_loadpath (filename)))
       error ("hgload: unable to locate file %s", filename);
     endif
   endif
-
+    
   ## Load the handle
   try
     stmp = load (filename, "s_oct40");
   catch
     error ("hgload: could not load hgsave-formatted object in file %s", filename);
   end_try_catch
 
   h = struct2hdl (stmp.s_oct40);
